File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
import math

def solution(n, words):
    l=len(words)
    exist=[words[0]]
    index=-1
    for i in range(1,l):
        if words[i-1][-1]!=words[i][0] or words[i] in exist:
            index=i
            break
        else:
            exist.append(words[i])


    if index==-1:
        return [0,0]
    return [index%n+1,index//n+1]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let l=words.length;
    let exist=[words[0]];
    let index=-1;
    for(let i=1;i<l;i++){
        if(words[i-1][words[i-1].length-1]!==words[i][0]||exist.includes(words[i])){
            index=i;
            break;
        }else
            exist.push(words[i]);
    }
    return index===-1?[0,0]:[index%n+1,~~(index/n)+1]
}